public/components/datasources/components/manage/accelerations/acceleration_operation.tsx

```diff
@@ -0,0 +1,102 @@
+import { DirectQueryLoadingStatus } from '../../../../../../common/types/explorer';
+import {
+  AccelerationActionType,
+  generateAccelerationOperationQuery,
+  getAccelerationName,
+} from './utils/acceleration_utils';
+
+export const useAccelerationOperation = (dataSource: string) => {
+  const { startLoading, stopLoading, loadStatus } = useDirectQuery();
+  const { setToast } = useToast();
+  const [isOperating, setIsOperating] = useState(false);
+  const [operationSuccess, setOperationSuccess] = useState(false);
+  const [accelerationToOperate, setAccelerationToOperate] = useState<CachedAcceleration | null>(
+    null
+  );
+  const [operationType, setOperationType] = useState<AccelerationActionType | null>(null);
+  const [currentStatus, setCurrentStatus] = useState<DirectQueryLoadingStatus | null>(null);
+
+  useEffect(() => {
+    if (!accelerationToOperate || !operationType || loadStatus === currentStatus) return;
+
+    const displayAccelerationName = getAccelerationName(
+      accelerationToOperate.indexName,
+      accelerationToOperate,
+      dataSource
+    );
+
+    let operationInProgressMessage = '';
+    let operationSuccessMessage = '';
+    let operationFailureMessage = '';
+
+    switch (operationType) {
+      case 'delete':
+        operationInProgressMessage = `Deleting acceleration: ${displayAccelerationName}`;
+        operationSuccessMessage = `Successfully deleted acceleration: ${displayAccelerationName}`;
+        operationFailureMessage = `Failed to delete acceleration: ${displayAccelerationName}`;
+        break;
+      case 'vacuum':
+        operationInProgressMessage = `Vacuuming acceleration: ${displayAccelerationName}`;
+        operationSuccessMessage = `Successfully vacuumed acceleration: ${displayAccelerationName}`;
+        operationFailureMessage = `Failed to vacuum acceleration: ${displayAccelerationName}`;
+        break;
+      case 'sync':
+        operationInProgressMessage = `Syncing acceleration: ${displayAccelerationName}`;
+        operationSuccessMessage = `Successfully synced acceleration: ${displayAccelerationName}`;
+        operationFailureMessage = `Failed to sync acceleration: ${displayAccelerationName}`;
+        break;
+    }
+
+    if (loadStatus === DirectQueryLoadingStatus.SCHEDULED) {
+      setIsOperating(true);
+      setToast(operationInProgressMessage, 'success');
+    } else if (loadStatus === DirectQueryLoadingStatus.SUCCESS) {
+      setIsOperating(false);
+      setAccelerationToOperate(null);
+      setOperationSuccess(true);
+      setToast(operationSuccessMessage, 'success');
+    } else if (loadStatus === DirectQueryLoadingStatus.FAILED) {
+      setIsOperating(false);
+      setOperationSuccess(false);
+      setToast(operationFailureMessage, 'danger');
+    }
+
+    setCurrentStatus(loadStatus);
+  }, [loadStatus, setToast, accelerationToOperate, dataSource, operationType, currentStatus]);
```

Comment:
Do we need this state? Feel like we could just have this `useEffect()` depend on the `loadStatus` from the `useDirectQuery()` hook. Our logic for these types of loading operations are usually something like:
1. `startLoading()`, set some state that indicates that the component is loading
2. Check if the component is done loading, i.e., `DirectQueryLoadingStatus.SUCCESS`. If so, perform success operations.
3. Check if the component failed, i.e., `DirectQueryLoadingStatus.FAILED`. If so, perform failure operations.

A very simple example of this can be found [here](https://github.com/opensearch-project/dashboards-observability/blob/main/public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/selector_helpers/load_databases.tsx#L32-L43). I think that could reduce the complexity, since checking we could perform the operations for `loadStatus === DirectQueryLoadingStatus.SCHEDULED` once we start loading, since it's guaranteed to happen.

Also, we probably don't need this many dependencies in the array.

Reply:
I was following the the example you mentioned. However, since our case required us to handle the toast pop up by using all these dependencies, so it became more complicated. For example, first, we need the `dataSource` and and `accelerationToOperate` to setup the displayed index name on the toast msg correctly. After that we also need the `operationType` for toast msgs. In addition, since we are bonding our toast msg with these status, and once the `use_ polling` class start fetching status by `startLoading`, it will keep fetching until it is `SUCCESS` or `FAILED` which means the logic here:
``` javascript
    if (loadStatus === DirectQueryLoadingStatus.SCHEDULED) {
      setIsOperating(true);
      setToast(operationInProgressMessage, 'success');
```
The above toast will keep being showed since the `DirectQueryLoadingStatus.SCHEDULED` will be there for a while. So the check of the `currentStatus` is guaranteed that the toast msg will only be poped once if there is no status change.

Reply:
> For example, first, we need the dataSource and and accelerationToOperate to setup the displayed index name on the toast msg correctly. 
Got it. `accelerationToOperate` is consistently changing, and the `useEffect()` should run once it changes since the acceleration will be set. `datasource` is initialized when the hook is initialized, so it won't be changing and thus, I think it's not necessary in the dependency array. You can test this on your side.
> So the check of the currentStatus is guaranteed that the toast msg will only be poped once if there is no status change.
Makes sense. I think my main comment was to try and avoid unnecessary runs of `useEffect()`. If this is what works, looks good to me.

Reply:
> datasource is initialized when the hook is initialized, so it won't be changing and thus, I think it's not necessary in the dependency array. 

Good catch! Fixed.